File: packages/expo-camera/android/src/main/java/expo/modules/camera/tasks/BarCodeScannerAsyncTaskDelegate.java
package expo.modules.camera.tasks;

import org.unimodules.core.interfaces.CodedThrowable;
import org.unimodules.interfaces.barcodescanner.BarCodeScannerResult;

public interface BarCodeScannerAsyncTaskDelegate {
  void onBarCodeScanned(BarCodeScannerResult barCode);
  void onBarCodeScanError(CodedThrowable throwable);
  void onBarCodeScanningTaskCompleted();
}

File: android/versioned-abis/expoview-abi45_0_0/src/main/java/abi45_0_0/expo/modules/camera/tasks/BarCodeScannerAsyncTaskDelegate.kt
package abi45_0_0.expo.modules.camera.tasks

import abi45_0_0.expo.modules.interfaces.barcodescanner.BarCodeScannerResult

interface BarCodeScannerAsyncTaskDelegate {
  fun onBarCodeScanned(barCode: BarCodeScannerResult)
  fun onBarCodeScanningTaskCompleted()
}
